cmd/influxd/config.go

```diff
@@ -197,6 +197,32 @@ func (c *Config) BrokerURL() *url.URL {
 	}
 }
 
+// BrokerDir returns the data directory to start up in and does home directory expansion if necessary.
+func (c *Config) BrokerDir() string {
+	path := c.Broker.Dir
+	u, _ := user.Current()
+	hd := u.HomeDir + "/"
+
+	// if path starts wiht ~/ we need to expand it
+	if path[:2] == "~/" {
```

Comment:
Whoa, why is this necessary? You mean if someone passes in a path like this, the OS doesn't expand it? Do you actually need this in practise?